core/src/main/java/edu/wpi/grip/core/operations/composite/PublishVideoOperation.java

```diff
@@ -10,34 +10,47 @@
 import com.google.common.collect.ImmutableList;
 
 import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
-
-import org.bytedeco.javacpp.BytePointer;
-import org.bytedeco.javacpp.IntPointer;
-
-import java.io.DataInputStream;
-import java.io.DataOutputStream;
-import java.io.IOException;
-import java.net.ServerSocket;
-import java.net.Socket;
+import edu.wpi.cscore.CameraServerJNI;
+import edu.wpi.cscore.CvSource;
+import edu.wpi.cscore.MjpegServer;
+import edu.wpi.cscore.VideoMode;
+import edu.wpi.first.wpilibj.networktables.NetworkTable;
+import edu.wpi.first.wpilibj.tables.ITable;
+
+import org.bytedeco.javacpp.opencv_core;
+import org.opencv.core.Mat;
+
+import java.lang.reflect.Field;
+import java.util.Deque;
+import java.util.LinkedList;
 import java.util.List;
 import java.util.logging.Level;
 import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
-import static org.bytedeco.javacpp.opencv_core.Mat;
-import static org.bytedeco.javacpp.opencv_imgcodecs.CV_IMWRITE_JPEG_QUALITY;
-import static org.bytedeco.javacpp.opencv_imgcodecs.imencode;
+import static org.bytedeco.javacpp.opencv_core.CV_8S;
+import static org.bytedeco.javacpp.opencv_core.CV_8U;
 
 /**
  * Publish an M-JPEG stream with the protocol used by SmartDashboard and the FRC Dashboard.  This
  * allows FRC teams to view video streams on their dashboard during competition even when GRIP has
- * exclusive access to the camera.  In addition, an intermediate processed image in the pipeline
- * could be published instead. Based on WPILib's CameraServer class:
- * https://github.com/robotpy/allwpilib/blob/master/wpilibj/src/athena/java/edu/wpi/first/wpilibj
- * /CameraServer.java
+ * exclusive access to the camera. Uses cscore to host the image streaming server.
  */
 public class PublishVideoOperation implements Operation {
 
   private static final Logger logger = Logger.getLogger(PublishVideoOperation.class.getName());
+
+  static {
+    try {
+      // Loading the CameraServerJNI class will load the appropriate platform-specific OpenCV JNI
+      CameraServerJNI.getHostname();
+    } catch (Throwable e) {
+      logger.log(Level.SEVERE, "CameraServerJNI load failed! Exiting", e);
+      System.exit(31);
```

Comment:
This should set a flag and make the operation throw an exception in `perform()` instead of suddenly exiting the entire application

Reply:
Yea, we want to limit the number crashes that don't display the crash dialog.

Reply:
Theres actually nothing to catch, because internally in the JNI loading classes we call System.exit(1). There would be no exception ever caught here. It would be impossible to catch.

Reply:
After a little discussion, we will leave the IO Exception that the static constructor can throw in flight, that way its catchable. That will be a change on our side. We will do the same thing to ntcore as well.